src/Raven.Server/Documents/ETL/Providers/AI/AiEtl.cs

```diff
@@ -0,0 +1,261 @@
+        var embeddingsObjectDjv = new DynamicJsonValue();
+
+        var dja = new DynamicJsonArray();
+        
+        dja.Add(attachmentGuid);
+        
+        // todo handle existing array
+        embeddingsObjectDjv[fieldName] = dja;
+            
+        documentDjv[Configuration.Name] = embeddingsObjectDjv;
+
+        using (var ctx = JsonOperationContext.ShortTermSingleUse())
+        {
+            var bjro = ctx.ReadObject(documentDjv, "doc");
+
+            var cmd = new MergedPutCommand(bjro, newDocumentId, null, Database);
+
+            Database.TxMerger.EnqueueSync(cmd);
+        }
+    }
+
+    private void CreateNewPrivateDocument(string textValue, float[] embeddingValue, DocumentsOperationContext context, out string attachmentGuid)
+    {
+        var hash = $"hash({textValue})";
+        var newDocumentId = GetPrivateDocumentId(hash);
+        
+        var documentDjv = new DynamicJsonValue { ["Id"] = newDocumentId, ["@metadata"] = new DynamicJsonValue() { ["@collection"] = "@embeddings" } };
+        
+        attachmentGuid = Guid.NewGuid().ToString();
+
+        documentDjv[textValue] = attachmentGuid;
+        
+        var embedding = GenerateEmbeddings.FromText(context.Allocator, VectorOptions.DefaultText, textValue).GetEmbedding().ToArray();
+        
+        using (var ctx = JsonOperationContext.ShortTermSingleUse())
+        {
+            var bjro = ctx.ReadObject(documentDjv, "doc");
+
+            var cmd = new MergedPutEmbeddingCommand(bjro, newDocumentId, null, new Dictionary<string, byte[]>() { { attachmentGuid, embedding } }, Database);
+
+            Database.TxMerger.EnqueueSync(cmd);
+        }
+    }
+
+    private static string GetPublicDocumentId(string originalDocumentId)
+    {
+        return $"{originalDocumentId}/embeddings";
+    }
+    
+    private string GetPrivateDocumentId(string hash)
+    {
+        return $"embeddings/{Configuration.Name}/{hash}";
+    }
+
+    protected override EtlStatsScope CreateScope(EtlRunStats stats)
+    {
+        return new EtlStatsScope(stats);
+    }
+
+    protected override bool ShouldFilterOutHiLoDocument()
+    {
+        throw new System.NotImplementedException();
+    }
+
+    private ITextEmbeddingGenerationService CreateService(AiEtlConfiguration configuration)
+    {
+        var kernelBuilder = Kernel.CreateBuilder();
+
+        switch (configuration.AiConnectorType)
+        {
+            case AiConnectorType.OpenAi:
+                var openAiSettings = configuration.Connection.OpenAiSettings;
+
+                var apiKey = new ApiKeyCredential(openAiSettings.ApiKey);
+                var openAiOptions = new OpenAIClientOptions
+                {
+                    Endpoint = new Uri(openAiSettings.Endpoint),
+                    OrganizationId = openAiSettings.OrganizationId,
+                    ProjectId = openAiSettings.ProjectId,
+                    UserAgentApplicationId = $"RavenDB/{ServerVersion.FullVersion}/{nameof(AiEtl)}"
```

Comment:
I think that ServerVersion.Version is enough. Check RavenHttpClient

Reply:
Done